plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -303,6 +306,101 @@ public void testRegisterTableWithMetadataFile()
         assertUpdate(format("DROP TABLE %s", tableName));
     }
 
+    @Test
+    public void testUnregisterTable()
+    {
+        String tableName = "test_unregister_table_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT 1 a", 1);
+        String tableLocation = getTableLocation(tableName);
+
+        assertUpdate("CALL system.unregister_table(CURRENT_SCHEMA, '" + tableName + "')");
+        assertQueryFails("SELECT * FROM " + tableName, ".* Table .* does not exist");
+
+        assertUpdate("CALL iceberg.system.register_table(CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+        assertQuery("SELECT * FROM " + tableName, "VALUES 1");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testUnregisterBrokenTable()
+    {
+        String tableName = "test_unregister_broken_table_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT 1 a", 1);
+        String tableLocation = getTableLocation(tableName);
+        String newTableLocation = tableLocation + "_new";
+
+        // Break the table by moving metadata files and register it in the new location
+        moveDirectory(tableLocation, newTableLocation);
```

Comment:
You cannot move Iceberg files & re-register, since metadata/manifests contain absolute paths.

For unregister broken table test, let's just delete the files from the storage, and see that unregister_table still works (=> table does not exist)

Reply:
> You cannot move Iceberg files & re-register, since metadata/manifests contain absolute paths.

I believe we **_can_**, but the table will be broken because `register_table` procedure doesn't verify those paths during the execution. This is a possible situation users may face by their mistake. 

> For unregister broken table test, let's just delete the files from the storage, and see that unregister_table still works (=> table does not exist)

👍